frontend/src/metabase/common/components/FilterPicker/FilterPicker.tsx

```diff
@@ -60,6 +64,16 @@ export function FilterPicker({
     onClose?.();
   };
 
+  const checkItemIsSelected = useCallback(
+    (item: ColumnListItem | SegmentListItem) => {
+      return !!(
+        typeof filterIndex === "number" &&
```

Comment:
nit: `filterIndex != null`. `typeof` checks makes me think if we expect other types, e.g. a string